### PR TITLE
Diluc and Kaeya Solution

### PR DESCRIPTION
**Pull request by SansPapyrus683**
Do I need to write some small clarifications? I feel like the original editorial is good enough.

All solutions have been tested.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
god exists only to spite me

File: solutions/cf-1536C.mdx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		int k_num = 0;
+		vector<int> max_pref_chunks;
+		std::unordered_map<int, std::unordered_map<int, int>> prev_ratios;
+		for (char c : str) {
+			if (c == 'D') {
+				d_num++;
+			} else if (c == 'K') {
+				k_num++;
+			}
+			int gcd = std::__gcd(d_num, k_num);
```
Comment (maggie-j-liu) [inline solutions/cf-1536C.mdx:45]: you can use gcd from the numeric header in c++17

File: solutions/cf-1536C.mdx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+using std::vector;
+
+// https://codeforces.com/problemset/problem/1536/C (input ommited due to length)
+int main() {
+	int test_num;
+	std::cin >> test_num;
+	for (int t = 0; t < test_num; t++) {
+		int len;
+		std::cin >> len;
+		vector<char> str(len);
```
Comment (maggie-j-liu) [inline solutions/cf-1536C.mdx:30]: why can't you just use a string instead of a char vector?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/cf-1536C.mdx [hunk 3]
```diff
@@ -0,0 +1,145 @@
+---
+id: cf-1427G
```
Comment (maggie-j-liu) [inline solutions/cf-1536C.mdx:2]: wrong id? you also need to change kind to internal in the .problems.json file

**Comment by maggie-j-liu** [comment]
> Do I need to write some small clarifications? I feel like the original editorial is good enough.

I think the editorial is good enough; but some comments in your code could help

File: solutions/cf-1536C.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/cf-1536C.mdx:30]: don't wanna put an extra #include

File: solutions/cf-1536C.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/cf-1536C.mdx:45]: don't wanna put an extra #include

File: solutions/cf-1536C.mdx (hunk 2 above)
Comment (maggie-j-liu) [reply solutions/cf-1536C.mdx:30]: I think the code would look cleaner and more straightforward with a string, but up to you

On an unrelated note, I think that strings usually do work without `#include <string>` but this is probably also implementation dependent

File: solutions/cf-1536C.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		int k_num = 0;
+		vector<int> max_pref_chunks;
+		std::unordered_map<int, std::unordered_map<int, int>> prev_ratios;
+		for (char c : str) {
+			if (c == 'D') {
+				d_num++;
+			} else if (c == 'K') {
+				k_num++;
+			}
+			// get the actual ratio by dividing both quantites by the gcd
```
Comment (maggie-j-liu) [inline solutions/cf-1536C.mdx:44]:
```suggestion
			// get the simplified ratio by dividing both quantities by the gcd
```
same for java, python comments

File: solutions/cf-1536C.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+		int k_num = 0;
+		vector<int> max_pref_chunks;
+		std::unordered_map<int, std::unordered_map<int, int>> prev_ratios;
+		for (char c : str) {
+			if (c == 'D') {
+				d_num++;
+			} else if (c == 'K') {
+				k_num++;
+			}
+			// get the simplified ratio by dividing both quantites by the gcd
```
Comment (maggie-j-liu) [inline solutions/cf-1536C.mdx:44]:
```suggestion
			// get the simplified ratio by dividing both quantities by the gcd
```
typo... fix the java and python comments too

Comment (SansPapyrus683) [reply solutions/cf-1536C.mdx:44]: lmao

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/cf-1536C.mdx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+				d_num++;
+			} else if (c == 'K') {
+				k_num++;
+			}
+			// get the simplified ratio by dividing both quantities by the gcd
+			int gcd = std::__gcd(d_num, k_num);
+			int d_ratio = d_num / gcd;
+			int k_ratio = k_num / gcd;
+			// add the ratio to the records and record the max prefix chunk amount
+			max_pref_chunks.push_back(++prev_ratios[d_ratio][k_ratio]);
```
Comment (dongliu0426) [inline solutions/cf-1536C.mdx:49]: why not just `cout  << ++prev_ratios[d_ratio][k_ratio]` here

Comment (SansPapyrus683) [reply solutions/cf-1536C.mdx:49]: whitespace is an absolute prick

**Pull request merged**